token-swap/program/src/math.rs

```diff
@@ -0,0 +1,265 @@
+//! Math utilities for swap calculations
+
+/// The representation of the number one as a precise number
+pub const ONE: u128 = 10_000_000_000;
+
+/// Maximum weight for token in swap
+pub const MAX_WEIGHT: u8 = 100;
+/// Minimum weight for token in swap
+pub const MIN_WEIGHT: u8 = 1;
+
+/// Struct encapsulating a fixed-point number that allows for decimal calculations
+#[derive(Clone)]
+pub struct PreciseNumber {
+    /// Wrapper over the inner value, which is multiplied by ONE
+    pub value: u128,
+}
+
+impl PreciseNumber {
+    const ONE: Self = Self { value: ONE };
+
+    const ROUNDING_CORRECTION: u128 = ONE / 2;
+    const POW_PRECISION: u128 = 100;
+    const APPROXIMATION_ITERATIONS: u64 = 100_000;
```

Comment:
What do the instruction counts on Swap execution look like with this limit?

Reply:
Let me add a perf monitor as you've recommended and get a sense of it.  These numbers were tuned for accuracy based on the unit tests, but could potentially be relaxed to get under any thresholds.